contracts/Trustcoin.sol

```diff
@@ -80,26 +76,33 @@ contract Trustcoin is OutgoingMigrationTokenInterface, ERC20TokenInterface, Safe
   }
 
   // See ERC20
-  // NOTE: this message is vulnerable and is placed here only to follow the ERC20 standard.
+  // NOTE: this method is vulnerable and is placed here only to follow the ERC20 standard.
   // Before using, please take a look at the better compareAndApprove below.
   function approve(address _spender, uint256 _value) external returns (bool) {
     return doApprove(_spender, _value);
   }
 
   // A vulernability of the approve method in the ERC20 standard was identified by
-  // Mikhail Vladimirov and Dmitry Khovratovich in http://bit.ly/2obzyE7 .
+  // Mikhail Vladimirov and Dmitry Khovratovich in this Google Doc:
+  // https://docs.google.com/document/d/1YLPtQxZu1UAvO9cZ1O2RPXBbT0mooh4DYKjA_jp-RLM
   // It's better to use this method which is not susceptible to over-withdrawing by the approvee.
   //
-  // _oldValue is the previous value approved, which can be retrieved with
-  //  allowance(msg.sender, _spender).
-  function compareAndApprove(address _spender, uint256 _currentValue, uint256 _newValue)
-      external returns(bool) {
+  /// @param _oldValue The previous value approved, which can be retrieved with allowance(msg.sender, _spender)
+  /// @return bool Whether the approval was a success (see ERC20's `approve`)
+  function compareAndApprove(address _spender, uint256 _currentValue, uint256 _newValue) external returns(bool) {
     if (allowed[msg.sender][_spender] != _currentValue) {
       return false;
     }
     doApprove(_spender, _newValue);
   }
 
+  // INTERNAL FUNCTIONS
```

Comment:
Matter of taste, but I rather that internal functions are at the bottom, so that readers who want to use this contract can easily figure out its API without getting confused by implementation issues.